File: Python/326.Power of Three.py
#判断给定数是否是3的次方
class Solution(object):
    def isPowerOfThree(self, n):
        while n>8:
            if n%3==0:
                n = n//3
            else:   return False
        return n==1 or n==3

s = Solution()
print(s.isPowerOfThree(19684))

File: C++/Math/326.Power of Three.cpp
/*
判断一个数是不是3的乘方。
如何不使用循环做？
*/
/*
思路：
*/
#include <iostream>
using namespace std;

class Solution {
public:
	bool isPowerOfThree(int n) {
    	// 1162261467 is 3^19,  3^20 is bigger than int  
    	return ( n>0 &&  1162261467%n==0);

    	//return fmod(log10(n)/log10(3), 1)==0;
    }
};
int main()
{
	Solution s;
	cout << s.isPowerOfThree(81) << endl;
}

